### DISCUSSION TITLE
What is the "Async Submit Validation" code example demonstrating?

### DISCUSSION THREAD

**Discussion by illright** [react-hook-form/react-hook-form#9694] [Q&A]
I'm looking for guidance on how to implement displaying server-side errors. I haven't found much in the documentation (except just the API page of `setError`, but nothing there says that this is the recommended way of handling submit errors), so I decided to check code examples. That's when I found the [Async Submit Validation](https://codesandbox.io/s/react-hook-form-async-submit-validation-hy9oe?file=/src/index.js:509-515) code example.

It has the following code:
```tsx
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = useForm();
  const onSubmit = async (data) => {
    await sleep(2000);
    if (data.username === "bill") {
      alert(JSON.stringify(data));
    } else {
      alert("There is an error");
    }
  };

  console.log(errors);
  
  // ...
  
  <div style={{ color: "red" }}>
    {Object.keys(errors).length > 0 &&
      "There are errors, check your console."}
  </div>
```

The thing is, I'm not sure why the `errors` object is referenced at all, since there is no validation in the example and the `onSubmit` callback doesn't call `setError`. I haven't found a way to see the `There are errors, check your console.` message printed. And the `onSubmit` method doesn't really demonstrate nice error handling, it just `alert`s the "backend response", which is something I would never do in real code.

Could you please update this code example to actually include the recommended way of displaying backend errors to the user?

**Comment by leapful** [comment]
@illright 

Thank you for your feedback. It should have `setError` to set custom error message for this case.

@bluebill1049 

Can we update the example to something like this ?

https://codesandbox.io/s/react-hook-form-async-submit-validation-forked-tiquz6?file=/src/index.js

**Reply by bluebill1049** [reply]
Great timing for this: https://github.com/react-hook-form/react-hook-form/discussions/9691

**Reply by leapful** [reply]
I just have thought about the new RFC global/server error too :D

**Reply by bluebill1049** [reply]
Happy new year @leapful! Wish you all the best for 2023.
